src/java.base/share/classes/javax/crypto/EncryptedPrivateKeyInfo.java

```diff
@@ -304,15 +315,167 @@ private PKCS8EncodedKeySpec getKeySpecImpl(Key decryptKey,
         }
     }
 
+    /**
+     * Returns an {@code EncryptedPrivateKeyInfo} from a given PrivateKey.
+     * A valid password-based encryption (PBE) algorithm and password must be
+     * specified.  AlgorithmParameterSpec, {@code params}, will use the
+     * provider default if {@code null} is passed.  If {@code provider} is
+     * {@code null}, the provider will be selected through the default
+     * provider list.
+     * <p>
+     * The PBE algorithm string format details can be found in the
+     * <a href="{@docRoot}/../specs/security/standard-names.html#cipher-algorithms">
+     * Cipher section</a> of the Java Security Standard Algorithm Names
+     * Specification.
+     *
+     * @param key the PrivateKey object to encrypt.
+     * @param password the password used for generating the PBE key.
+     * @param algorithm the PBE encryption algorithm.
+     * @param params the parameters used with the PBE encryption.
+     * @param provider the Provider that will perform the encryption.
+     * @return an EncryptedPrivateKeyInfo.
+     * @throws IllegalArgumentException when an argument causes an
+     * initialization error.
+     * @throws SecurityException on a cryptographic errors.
```

Comment:
Oh, I didn't notice this before. Have we decided to repurpose `SecurityException` for this usage now that there will be no more Security Manager?

Reply:
I'm not sure if it was exclusively a Security Manager exception, but @seanjmullan had said previously that `SecurityException` was used elsewhere.

Reply:
It was used when `JarInputStream::read` encountered a wrong hash or signature. But, it was mainly used for permission violation.